### PR TITLE
EpochRewardSummary filter with FromEpoch, ToEpoch, MarketIDs and AssetIDs

### LINKED ISSUE #7304: Epoch reward summaries filter
# Feature Overview

**In order to** not overfetch data when using the epochRewardSummaries query
**We will** add a filter for assetId and marketId
**So that** I can fetch only rewards for single market or asset

# Acceptance Criteria
- It is possible to filter reward summaries by asset ID
- It is possible to filter reward summaries by market ID

# Examples (optional)

```
input RewardSummaryFilter {
  assetIds: [ID!]
  marketIds: [ID!]
}

Query {
  epochRewardSummaries(
    filter: RewardSummaryFilter,
    pagination:  ...
  ): [EpochRewardSummaryConnection!]!
}
```

### PR DESCRIPTION
**Pull request by zale144**
Resolves #7304

### CODE REVIEW

File: datanode/gateway/graphql/resolvers.go [hunk 1]
```diff
@@ -698,6 +703,10 @@ func fromPtr[T any](ptr *T) (ret T) {
 	return
 }
 
+func toPtr[T any](val T) *T {
```
Comment (wwestgarth) [inline datanode/gateway/graphql/resolvers.go:706]: I think we have this already in `lib/ptr/ptr.go` that you could pull in.

Comment (zale144) [reply datanode/gateway/graphql/resolvers.go:706]: Nice. Somehow I couldn't find it

**Review by edd — DISMISSED** [review]

**Review by edd — DISMISSED** [review]

**Review by peterbarrow — DISMISSED** [review]

**Comment by AnExsomnis** [comment]
@zale144 , are these 18 commits yours?

**Comment by zale144** [comment]
> @zale144 , the 18 commits include other people's commits as well.
> Maybe you should rebase after [4162ac9](https://github.com/vegaprotocol/vega/commit/4162ac9db3da606f8369189db4679c133bec499d)?

@AnExsomnis Yeah, at some point a rebase happened and now all these commits have different hashes, so git thinks they are different commits, while they are the same, only the history is different. Normally I just merge the target branch into mine, in order to avoid resolving conflicts per each commit, and to avoid messing up the history

**Review by AnExsomnis — APPROVED** [review]

**Review by edd — APPROVED** [review]
👍 Still no-one using this query internally so while it's a breaking GraphQL schema change, it won't break anyone's code

**Pull request merged**